app/src/main/java/emu/skyline/input/InputHandler.kt

```diff
@@ -275,9 +246,9 @@ class InputHandler(private val inputManager : InputManager, private val preferen
 
         motionSensor.deltaTimestamp = event.timestamp.toULong() - motionSensor.timestamp
         motionSensor.timestamp = event.timestamp.toULong()
-        setMotionState(0, 0, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x5C).order(ByteOrder.LITTLE_ENDIAN)))
-        setMotionState(0, 1, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x5C).order(ByteOrder.LITTLE_ENDIAN)))
-        setMotionState(0, 2, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x5C).order(ByteOrder.LITTLE_ENDIAN)))
+        setMotionState(0, 0, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)))
+        setMotionState(0, 1, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)))
+        setMotionState(0, 2, motionSensor.writeToByteBuffer(ByteBuffer.allocateDirect(0x28).order(ByteOrder.LITTLE_ENDIAN)))
```

Comment:
To avoid continuously allocating ByteBuffers at every motion event, you should add one `ByteBuffer motionDataBuffer`, member to the `InputHandler` class initialised with correct size and ordering, and then here do:
```suggestion
        setMotionState(0, 0, motionSensor.writeToByteBuffer(motionDataBuffer)))
        setMotionState(0, 1, motionSensor.writeToByteBuffer(motionDataBuffer)))
        setMotionState(0, 2, motionSensor.writeToByteBuffer(motionDataBuffer)))
```